src/config.ts

```diff
@@ -0,0 +1,230 @@
+import { log } from "./utils/logger";
+
+import {
+  createClusterRedisClient,
+  createSimpleRedisClient
+} from "./utils/redis";
+
+// Without this the environment variables loaded by dotenv aren't available in
+// this file.
+dotenv.config();
+
+// Server port.
+const DEFAULT_SERVER_PORT = "80";
+export const SERVER_PORT: number = parseInt(
+  process.env.PORT || DEFAULT_SERVER_PORT,
+  10
+);
+
+export const ENV = getNodeEnvironmentFromProcessEnv(process.env);
+
+// Default cache control max-age value is 5 minutes
+const DEFAULT_CACHE_MAX_AGE_SECONDS: string = "300";
+
+// Resolve cache control default max-age value
+export const CACHE_MAX_AGE_SECONDS: number = parseInt(
+  process.env.CACHE_MAX_AGE_SECONDS || DEFAULT_CACHE_MAX_AGE_SECONDS,
+  10
+);
+
+// Private key used in SAML authentication to a SPID IDP.
+const samlKey = () => {
+  return readFile(
+    process.env.SAML_KEY_PATH || "./certs/key.pem",
+    "SAML private key"
+  );
+};
+export const SAML_KEY = samlKey();
+
+// Public certificate used in SAML authentication to a SPID IDP.
+const samlCert = () => {
+  return readFile(
+    process.env.SAML_CERT_PATH || "./certs/cert.pem",
+    "SAML certificate"
+  );
+};
+
+export const SAML_CERT = samlCert();
+
+// SAML settings.
+const SAML_CALLBACK_URL =
+  process.env.SAML_CALLBACK_URL ||
+  "http://italia-backend/assertionConsumerService";
+const SAML_ISSUER = process.env.SAML_ISSUER || "https://spid.agid.gov.it/cd";
+const DEFAULT_SAML_ATTRIBUTE_CONSUMING_SERVICE_INDEX = "1";
+const SAML_ATTRIBUTE_CONSUMING_SERVICE_INDEX: number = parseInt(
+  process.env.SAML_ATTRIBUTE_CONSUMING_SERVICE_INDEX ||
+    DEFAULT_SAML_ATTRIBUTE_CONSUMING_SERVICE_INDEX,
+  10
+);
+const DEFAULT_SAML_ACCEPTED_CLOCK_SKEW_MS = "-1";
+const SAML_ACCEPTED_CLOCK_SKEW_MS = parseInt(
+  process.env.SAML_ACCEPTED_CLOCK_SKEW_MS ||
+    DEFAULT_SAML_ACCEPTED_CLOCK_SKEW_MS,
+  10
+);
+const DEFAULT_SPID_AUTOLOGIN = "";
+const SPID_AUTOLOGIN = process.env.SPID_AUTOLOGIN || DEFAULT_SPID_AUTOLOGIN;
+const DEFAULT_SPID_TESTENV_URL = "https://spid-testenv2:8088";
+const SPID_TESTENV_URL =
+  process.env.SPID_TESTENV_URL || DEFAULT_SPID_TESTENV_URL;
+
+// Register the spidStrategy.
+export const IDP_METADATA_URL = getRequiredENVVar("IDP_METADATA_URL");
+export function generateSpidStrategy(): Promise<SpidStrategy> {
+  return spidStrategy(
+    SAML_KEY,
+    SAML_CERT,
+    SAML_CALLBACK_URL,
+    SAML_ISSUER,
+    SAML_ACCEPTED_CLOCK_SKEW_MS,
+    SAML_ATTRIBUTE_CONSUMING_SERVICE_INDEX,
+    SPID_AUTOLOGIN,
+    SPID_TESTENV_URL,
+    IDP_METADATA_URL
+  );
+}
+
+// Redirection urls
+const clientProfileRedirectionUrl =
+  process.env.CLIENT_REDIRECTION_URL || "/profile.html?token={token}";
+
+if (!clientProfileRedirectionUrl.includes("{token}")) {
+  log.error("CLIENT_REDIRECTION_URL must contains a {token} placeholder");
+}
+
+// Range IP allowed for notification.
+function decodeNotifyCIDR(): string & IPatternStringTag<string> {
```

Comment:
```suggestion
function decodeNotifyCIDR(): CIDR {
```

https://github.com/teamdigitale/io-ts-commons/blob/master/src/strings.ts#L129